RNWCPP/Chakra/ChakraJsiRuntime_core.cpp

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "ChakraJsiRuntime.h"
+#include "UnicodeConversion.h"
+
+#include <cxxreact/MessageQueueThread.h>
+
+// This file contains non-edge-mode (or win32) implementations.
+#if !defined(USE_EDGEMODE_JSRT)
```

Comment:
It would be more useful to have a define for whether using ChakraCore or Chakra, and have that derive whether to define USE_EDGEMODE_JSRT or not. 

ChakraCore debugging only applies to ChakraCore and not to Chakra, so this would follow the define of which is being used. I don't see a need to also define CHAKRART_HAS_DEBUGGER.

Reply:
I don't follow the split between ChakraJsiRuntime_core.cpp and ChakraJsiRuntime_edgemode.cpp.  ChakraCoreDebugger.h is only needed for ChakraCore but it is inside ChakraJsiRuntime_core.cpp.

Reply:
It might be more useful to treat Chakra and ChakraCore as two separate runtimes, but then have shared code for their implementations. Rather than just one ChakraJsiRuntime which can be compiled two ways it would be two separate (albeit similar) runtimes which could be chosen.